### ISSUE TITLE
Failed calling webhook "vingress.elbv2.k8s.aws": the server could not find the requested resource

### ISSUE THREAD

**Issue by tru-cstavares** [kubernetes-sigs/aws-load-balancer-controller#2479]
**Describe the bug**
We are trying to install an ingress but it fails with following error: 

```bash
Error from server (InternalError): error when creating "ingress.yml": Internal error occurred: failed calling webhook "vingress.elbv2.k8s.aws": the server could not find the requested resource
```

Our ingress is being created as follows (ingress.yaml):

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: myIngressgateway
  namespace: test
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS": 443}]'
    alb.ingress.kubernetes.io/actions.ssl-redirect: '443'
    alb.ingress.kubernetes.io/load-balancer-name: myIngressGateway
    alb.ingress.kubernetes.io/target-type: instance
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:eu-west-1:xxxxxxxxxxxx:certificate/xxxxxxxxxxxx
    alb.ingress.kubernetes.io/backend-protocol: HTTPS
  labels:
    app: myIngressGatewway
spec:
  defaultBackend:
    service:
      name: myIngressgateway
      port:
        number: 80
  rules:
    - http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: myIngressgateway
                port:
                  number: 443
```

I could successfully create the Ingress later editing the ValidatingWebhookConfiguration replacing v1 by v1beta1 however this is an workaround not the final solution. I had edited as follows:

```yaml
- admissionReviewVersions:
  - v1beta1
  clientConfig:
    caBundle: XXXXXXXXXX
    service:
      name: aws-load-balancer-webhook-service
      namespace: test
      path: /validate-networking-v1beta1-ingress  <-------------------------------------------- previous was /validate-networking-v1-ingress
      port: 443
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: vingress.elbv2.k8s.aws
  namespaceSelector: {}
  objectSelector: {}
  rules:
  - apiGroups:
    - networking.k8s.io
    apiVersions:
    - v1beta1 <-------------------------------------------- previous was v1
    operations:
    - CREATE
    - UPDATE
    resources:
    - ingresses
    scope: '*'
  sideEffects: None
  timeoutSeconds: 10
```

On aws-load-balancer-controller pod logs I could see the following success mesage.

```
{"level":"debug","ts":1643295482.1303256,"logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/validate-networking-v1beta1-ingress","UID":"32e32888-6e1c-45a9-8185-62e53e742397" (...)

{"level":"debug","ts":1643295482.1305826,"logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/validate-networking-v1beta1-ingress","code":200,"reason":"","UID":"32e32888-6e1c-45a9-8185-62e53e742397","allowed":true}
```

Could someone check why its failing without the workaround on ValidatingWebhookConfiguration?

**Steps to reproduce**
We use helm template (from this [repo](https://github.com/kubernetes-sigs/aws-load-balancer-controller/tree/v2.3.1/helm/aws-load-balancer-controller)) and save output to file and then kubectl apply of that file.

Then we apply the ingress.yaml (see above).

**Expected outcome**
Its expected that ingress would create with success.

**Environment**

* AWS Load Balancer controller version
aws-load-balancer-controller:v2.3.1
* Kubernetes version
v1.20.7
* Using EKS (yes/no), if so version?
Yes, v1.20

**Comment by haouc** [comment]
Hey @tru-cstavares, yes, we are aware of this issue if users `helm install` from this repo/helm or eks-charts/stable/aws-load-balancer-controller directly.. We are working on it.

**Comment by kishorj** [comment]
@tru-cstavares, helm template from this repo does not work with aws-load-balancer-controller:v2.3.x releases. You either need to build the image from main branch, or wait for the v2.4.0 to be released. Please use the eks-charts repo as pointed by @haouc

**Comment by tru-cstavares** [comment]
Hi @haouc, @kishorj thanks for your prompt reply. I was not aware of that. Is this written somewhere? 
Regarding v2.4.0, when is planned to be released?

**Comment by kishorj** [comment]
we've just released v2.4.0. closing the issue.
